pkgs/data/fonts/apple-fonts/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib, stdenv, fetchurl, unzip, p7zip }:
+
+stdenv.mkDerivation rec {
+  pname = "apple-fonts";
+  version = "1.0.3";
+
+  pro = fetchurl {
+    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Pro.dmg";
+    sha256 = "sha256-nkuHge3/Vy8lwYx9z+pvsQZfzrNIP4K0OutpPl4yXn0=";
+  };
+
+  compact = fetchurl {
+    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Compact.dmg";
+    sha256 = "sha256-+Q4HInJBl3FLb29/x9utf7A55uh5r79eh/7hdQDdbSI=";
+  };
+
+  mono = fetchurl {
+    url = "https://devimages-cdn.apple.com/design/resources/download/SF-Mono.dmg";
+    sha256 = "sha256-pqkYgJZttKKHqTYobBUjud0fW79dS5tdzYJ23we9TW4=";
+  };
+
+  ny = fetchurl {
+    url = "https://devimages-cdn.apple.com/design/resources/download/NY.dmg";
+    sha256 = "sha256-XOiWc4c7Yah+mM7axk8g1gY12vXamQF78Keqd3/0/cE=";
+  };
+
+  nativeBuildInputs = [ p7zip ];
+
+  sourceRoot = ".";
+
+  dontUnpack = true;
+
+  installPhase = ''
+    7z x ${pro}
+    cd SFProFonts
+    7z x 'SF Pro Fonts.pkg'
+    7z x 'Payload~'
+    mkdir -p $out/fontfiles
+    mv Library/Fonts/* $out/fontfilesn
+    cd ..
+    7z x ${mono}
+    cd SFMonoFonts
+    7z x 'SF Mono Fonts.pkg'
+    7z x 'Payload~'
+    mv Library/Fonts/* $out/fontfiles
+    cd ..
+    7z x ${compact}
+    cd SFCompactFonts
+    7z x 'SF Compact Fonts.pkg'
+    7z x 'Payload~'
+    mv Library/Fonts/* $out/fontfiles
+    cd ..
+    7z x ${ny}
+    cd NYFonts
+    7z x 'NY Fonts.pkg'
+    7z x 'Payload~'
+    mv Library/Fonts/* $out/fontfiles
+    mkdir -p $out/usr/share/fonts/OTF $out/usr/share/fonts/TTF
+    mv $out/fontfiles/*.otf $out/usr/share/fonts/OTF
+    mv $out/fontfiles/*.ttf $out/usr/share/fonts/TTF
+    rm -rf $out/fontfiles
+  '';
+
+  meta = with lib; {
+    description = "Apple San Francisco, New York fonts";
+    homepage = "https://developer.apple.com/fonts/";
+    license = lib.licenses.unfree;
+    #maintainers = with maintainers; [ TheDuckWithAMonocle ];
```

Comment:
Please add yourself as a maintainer in a separate commit before adding the package.